unit/src/SplitFileNameTest.C

```diff
@@ -29,6 +29,6 @@ TEST(SplitFileName, validName)
 
   full = "valid.txt";
   split = MooseUtils::splitFileName(full);
-  EXPECT_EQ(split.first.compare("."), 0);
+  EXPECT_EQ(split.first.compare(""), 0);
```

Comment:
Let me try using `lexically_normal()` : https://www.godbolt.org/z/55xh751aj

Reply:
Yeah, I don;t think that's going to help: https://www.godbolt.org/z/axnT6YWWo
Explicitly normalizing an empty path results in an empty path. That goes against what the docs are saying IMO...

Reply:
But it looks like explicitly normalizing any other path string that should evaluate to "." does in fact evaluate to "."?

Can we take advantage of that to retain backwards compatibility?  Even if we're doing the `path` construction implicitly, we could always test for "" afterward, then manually change "" to "." if we find it.

I'd originally also worried about forwards compatibility, if we're interpreting the docs correctly and the implementations might catch up to us later, but looking at the C++17 standard (well, n4659, the closest draft you can get for free) I see the very first step in "30.10.4.11 - fs.def.normal.form" is "1. If the path is empty, stop."

Reply:
Wait, what's the conclusion here? Currently we respect "If the path is empty, stop". That's the C++ convention. But nobody say we need to switch to that. We could just maintain backward compatibility to the "MOOSE convention" instead.

Personally I'm leaning towards the implementation that results in less code. Even if that means a slight change in the path.

Reply:
I guess I'm satisfied either way, since MOOSE has such good CI coverage of it's downstream apps that we can hope to find any conflicts quickly.

If this was in libMesh I'd need more incentive than "less code" to avoid breaking downstream behavior without at least a period of "the old behavior is default, the new behavior is enabled with `configure --disable-deprecated`" to enable a smoother transition.